### ISSUE TITLE
Rectangular ROI selection

### ISSUE THREAD

**Issue by XanAstia** [fermiPy/fermipy#313]
Hi, 
I am currently doing an analysis of a region with fermipy but I would like to select a rectangular ROI rather than a square or circular ROI.
I am under the impression that this is not possible with the current version of fermipy, is that correct ?
Otherwise is it possible to add this feature ? Would you be available to give me some guidance for doing it myself ? 

Thank you for your help.

**Comment by tibaldo** [comment]
I also need this feature and I think it's rather straightforward to implement. I have a working version, still not thoroughly tested in my branch [rectangular-roi](https://github.com/tibaldo/fermipy/tree/rectangular_roi).
Would there be any interest to merge this into master once finalized?
Can someone please point me to the instructions to run the test routines? I'm afraid I didn't manage to find them.

**Comment by tibaldo** [comment]
@eacharles  helped me offline to figure out how to run the tests, and at this point I think I'm essentially done.

The key change is that the `npix` configuration parameter and the `_npix` attributes of GTAnalysis and GTBinnedAnalysis become tuples, that give the number of pixels along the x and y dimensions, respectively. All the underlying methods from GammaPy and the fermitools were in fact already designed to take separately the number of bins in the x and y dimensions, so I just had to tweak the code in a few places to make sure the right one was passed (before it was irrelevant since the maps were always square). If the user does not specify `npix` in the configuration file, but instead specifies `roiwidth` and `binsz` as customary in fermipy, everything works as before, i.e., the analysis region is a square with number of bins determined from `roiwidth`/`binsz` (rounded up).

The unit tests are passed. I also run a couple of analyses by hand to make sure everything was fine after my changes. Here I post an example which consists in an analysis of the Crab region using 1 month of Fermi data. This is the configuration file

```
data:
  evfile : FT1.fits
  scfile : FT2.fits

binning:
  binsz      : 0.1
  npix:
      - 100
      - 120
  binsperdec : 8

selection :
  ra : 83.6331
  dec : 23.0145
  emin : 100
  emax : 300000
  zmax : 90
  evclass : 128
  evtype : 3
  
gtlike:
  edisp : False
  irfs : 'P8R3_SOURCE_V2'

model:
  src_roiwidth : 15.0
  galdiff  : '$FERMI_DIFFUSE_DIR/gll_iem_v07.fits'
  isodiff  : '$FERMI_DIFFUSE_DIR/iso_P8R3_SOURCE_V2_v1.txt'
  catalogs : ['4FGL']
```

Note that for the sake of the test the map is not symmetric and the Crab is offset from the center.

Here are residual maps, with all the 4FGL sources in the model
![fit1_resid_pointsource_powerlaw_2 00_residmap_excess](https://user-images.githubusercontent.com/3046743/76309272-c10e1f00-62cc-11ea-917c-b8f70bda56b7.png)
and excluding the 3 sources associated with the Crab pulsar+nebula.
![fit1_residnoCrab_pointsource_powerlaw_2 00_residmap_excess](https://user-images.githubusercontent.com/3046743/76309321-daaf6680-62cc-11ea-9387-95889083fd5c.png)
The 4FGL model is rather good, and the excess when excluding the Crab sources appears at the right spot.

Then, I went on to delete the 3 sources associated with the Crab and compute a TS map.
![fit1_noCrab_pointsource_powerlaw_2 00_tsmap_sqrt_ts](https://user-images.githubusercontent.com/3046743/76309655-7e007b80-62cd-11ea-971d-62b22ff05c1d.png)
Again, there is a TS cluster centered on the Crab position.

Afterwards, I run `find_sources`. One source is found at a position consistent with the Crab nebula. Here are the residuals after this step.
![fit2_pointsource_powerlaw_2 00_residmap_excess](https://user-images.githubusercontent.com/3046743/76309520-38dc4980-62cd-11ea-89cb-cf40f30e06ca.png)
In this case there are still significant residuals around the Crab, because the simple power-law spectral model used cannot capture the complex spectrum of the pulsar+nebula.

Finally, I tested the newly found source for extension using the `extension` method. The source does not turn out to be significantly extended
![ps_j0534 5+2200_extension](https://user-images.githubusercontent.com/3046743/76309828-e2233f80-62cd-11ea-8e2d-c43700dc68bc.png)
and the 95% upper limit on the extension of 0.048 deg is larger that the extensions measured with Fermi/HESS.

In summary, everything looks alright. I'm glad to make a pull request if you want to include these changes in the master branch. I did not add anything so far to the test units to check the proper working of the asymmetric map mode but that can be easily done, I just need to understand how to add a file to fermipy-extra.

PS: in the process I ended up updating the 4FGL extended source archive because the version I got had corrupted files. Of course you don't need to integrate this if it has beed fixed elsewhere or if you think it's my own problem.

**Comment by eacharles** [comment]
The changes have been tested and merged in to the head
